From 19e20d6cd79b8d60963749b1556360394fa629b4 Mon Sep 17 00:00:00 2001
From: Sami Kyostila <skyostil@chromium.org>
Date: Mon, 14 Aug 2023 04:50:32 +0000
Subject: [PATCH] base/trace_event: Don't initialize Perfetto for testing

Disable the code for initializing Perfetto for testing. This saves
300+ KiB of binary size by removing the dependency on Perfetto's
in-process tracing service which isn't needed by production code.

Change-Id: I99c002b6ee50fe237c2aa7dd4943949a4faad793
patch-name: long-term-1050-Don-t-initialize-Perfetto-for-testing.patch
---
 base/trace_event/trace_event_impl.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/base/trace_event/trace_event_impl.cc b/base/trace_event/trace_event_impl.cc
index 599cb19de2..b46007b572 100644
--- a/base/trace_event/trace_event_impl.cc
+++ b/base/trace_event/trace_event_impl.cc
@@ -85,7 +85,9 @@ void TraceEvent::InitArgs(TraceArguments* args) {
 }
 
 void SetPerfettoInitializedForTesting() {
+#if 0
   g_perfetto_initialized_for_testing = true;
+#endif
 }
 
 bool IsPerfettoInitializedForTesting() {
@@ -93,6 +95,7 @@ bool IsPerfettoInitializedForTesting() {
 }
 
 void InitializeInProcessPerfettoBackend(perfetto::Platform* platform) {
+#if 0
   perfetto::TracingInitArgs init_args;
   init_args.platform = platform;
   init_args.backends = perfetto::BackendType::kInProcessBackend;
@@ -102,6 +105,7 @@ void InitializeInProcessPerfettoBackend(perfetto::Platform* platform) {
   init_args.disallow_merging_with_system_tracks = true;
   perfetto::Tracing::Initialize(init_args);
   TrackEvent::Register();
+#endif
 }
 
 }  // namespace base::trace_event
-- 
2.55.0.1082.g2b9226bbc0-goog

